bundles/org.openhab.binding.bluetooth/src/main/java/org/openhab/binding/bluetooth/discovery/internal/BluetoothDiscoveryProcess.java

```diff
@@ -227,18 +245,60 @@ public void onConnectionStateChange(BluetoothConnectionStatusNotification connec
         }
     }
 
-    private void tryToDiscoverNameIfMissing() throws InterruptedException {
-        BluetoothCharacteristic characteristic = device.getCharacteristic(GattCharacteristic.DEVICE_NAME.getUUID());
-        if (characteristic == null || device.getName() != null) {
-            return;
-        }
-        if (!device.readCharacteristic(characteristic)) {
-            logger.debug("Failed to aquire name for device {}", device.getAddress());
-            return;
-        }
-        if (!awaitNameDiscovery(1, TimeUnit.SECONDS)) {
-            logger.debug("Name discovery for device {} timed out", device.getAddress());
-        }
+    private void readDeviceInformationIfMissing() {
+        deviceInformationUUIDs.forEach(uuid -> {
+            BluetoothCharacteristic characteristic = device.getCharacteristic(uuid);
+            if (characteristic == null) {
+                logger.debug("Device '{}' doesn't support uuid '{}'", device.getAddress(), uuid);
+                return;
+            }
+            switch (characteristic.getGattCharacteristic()) {
+                case DEVICE_NAME:
+                    if (device.getName() != null) {
+                        return;
+                    }
+                    break;
+                case MODEL_NUMBER_STRING:
+                    if (device.getModel() != null) {
+                        return;
+                    }
+                    break;
+                case SERIAL_NUMBER_STRING:
+                    if (device.getSerialNumber() != null) {
+                        return;
+                    }
+                    break;
+                case HARDWARE_REVISION_STRING:
+                    if (device.getHwRevision() != null) {
+                        return;
+                    }
+                    break;
+                case FIRMWARE_REVISION_STRING:
+                    if (device.getFwRevision() != null) {
+                        return;
+                    }
+                    break;
+                case SOFTWARE_REVISION_STRING:
+                    if (device.getSwRevision() != null) {
+                        return;
+                    }
+                    break;
+                default:
+                    return;
+            }
+            if (!device.readCharacteristic(characteristic)) {
+                logger.debug("Failed to aquire uuid {} from device {}", uuid, device.getAddress());
+                return;
+            }
+            ongoingDevInfoUUID = Optional.of(uuid);
+            try {
+                if (!awaitInfoResponse(2, TimeUnit.SECONDS)) {
```

Comment:
Why was the timeout increased to 2 seconds? It should never take longer than a split second to communicate to a connected device. My use of 1 second was already excessive.

Reply:
There could be many request on the queue. Even 2 seconds might not be enough. Anyhow, I changed it back to 1 second.

Reply:
How many requests at once are you trying to take into account? I think that 10 requests at once would be the max to reasonably expect.

Reply:
Don't know, but example one airthings device service discovery take around 5 seconds (47 characteristics). I increased that time from 1 sec to 5 and it seems work rather well, but still some times I saw timeout (service discovery fails). Not sure what happens if there could be more than one airthings devices.

Reply:
I think that the double service discovery calls might be causing the timeout issues. I'm ok with you increasing the service discovery timeout to whatever value works for you. But the characteristic read/write timeouts should remain low.